core/src/main/java/com/sap/ai/sdk/core/AiCoreService.java

```diff
@@ -225,6 +182,6 @@ protected ApiClient buildApiClient(@Nonnull final Destination destination) {
    * @param resourceGroup the resource group of the deleted deployment, usually "default".
    */
   public void reloadCachedDeployments(@Nonnull final String resourceGroup) {
-    DEPLOYMENT_CACHE.resetCache(this, resourceGroup);
+    deploymentResolver.reloadDeployments(resourceGroup);
```

Comment:
(Minor)

I understand your PR didn't introduce it. But I'm not sure whether we should pollute this class with cache mechanics :)

Not sure whether double locking is necessary here.

Can we annotate it `@Beta`?

Reply:
Agreed. In general, currently `AiCoreService` is just re-exposing the API defined by `DeploymentResolver`. I was contemplating to just make `DeploymentResolver` public, getting rid of this duplication. But that would come at the downside that the usage gets a bit more verbose:

```java
new AiCoreService().getDestinationForDeploymentByModel(..)
new AiCoreService().getDeploymentResolver().getDestinationForDeploymentByModel(..)
```

No strong opinion. We could also mark all the `getDestinationForDeployment` as beta to allow for future changes..

Reply:
(marked the cache reload as beta now)